src/state/slices/foxEthSlice/foxEthSlice.ts

```diff
@@ -21,10 +22,10 @@ import {
 import { bn, bnOrZero } from 'lib/bignumber/bignumber'
 import { logger } from 'lib/logger'
 import { BASE_RTK_CREATE_API_CONFIG } from 'state/apis/const'
-import type { MarketDataState } from 'state/slices/marketDataSlice/marketDataSlice'
 import { marketData } from 'state/slices/marketDataSlice/marketDataSlice'
 
 import type { AssetsState } from '../assetsSlice/assetsSlice'
+import { selectMarketDataById } from '../selectors'
```

Comment:
tsc was being drunk when checking for `!marketDataState.crypto.byId[ethAssetId]`, somehow was losing the narrowing afterwards - this fixes it and gives us some granularity while at it